packages/tools/devtools/devtools-core/src/data-visualization/DataVisualization.ts

```diff
@@ -224,15 +247,20 @@ export class DataVisualizerGraph
 	 * Adds a visualizer node to the collection for the specified
 	 * {@link @fluidframework/shared-object-base#ISharedObject} if one does not already exist.
 	 */
-	private registerVisualizerForSharedObject(sharedObject: ISharedObject): FluidObjectId {
+	private registerVisualizerForSharedObject(
+		sharedObject: ISharedObject | DataObject,
+	): FluidObjectId {
 		if (!this.visualizerNodes.has(sharedObject.id)) {
 			// Create visualizer node for the shared object
-			const visualizationFunction =
-				this.visualizers[sharedObject.attributes.type] ?? visualizeUnknownSharedObject;
+			const visualizationFunction = isDataObject(sharedObject)
+				? visualizeDataObject
+				: (this.visualizers[sharedObject.attributes.type] ?? visualizeUnknownSharedObject);
 
 			const visualizerNode = new VisualizerNode(
-				sharedObject,
-				visualizationFunction,
+				// Typecasting `sharedObject` to `any` is necessary for `DataObject` visualization`, because the `root` property is inaccessbile (private).
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+				isDataObject(sharedObject) ? (sharedObject as any).root : sharedObject,
```

Comment:
Even if we have to type cast, doing it to the specific thing that we expect to be correct is better and avoids all the pitfalls of `any`. Or can we do `as VisualDataObject`?

```suggestion
				// Typecasting `sharedObject` is necessary for `DataObject` visualization because the `root` property is not exposed.
				isDataObject(sharedObject) ? (sharedObject as { readonly root: ISharedDirectory }).root : sharedObject,
```

Reply:
I think either way works. Typecasting it to `VisualDataObject` is concise and more readable. And I think `isDataObject()` is sufficient to determine the type of the node. 

On the other hand, `(sharedObject as { readonly root: ISharedDirectory }).root` is *wordier*, but gives more direct impression that the `root` must be `ISharedDirectory` and will be helpful when we are maintaining this in the future.

Reply:
BTW, the linter forces `unknown` conversion before hand:
- `(sharedObject as unknown as { readonly root: ISharedDirectory }).root`

Reply:
Ah, yeah, that makes sense. There isn't any overlap between the two type signatures. That still seems fine to me. The important thing is that we document what's going on.